src/main/java/com/yelp/nrtsearch/server/field/TextFieldDef.java

```diff
@@ -15,15 +15,47 @@
  */
 package com.yelp.nrtsearch.server.field;
 
+import com.yelp.nrtsearch.server.field.properties.PrefixQueryable;
 import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.PrefixQuery;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.lucene.document.Document;
 import org.apache.lucene.document.FieldType;
 import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
 
 /** Field class for 'TEXT' field type. */
-public class TextFieldDef extends TextBaseFieldDef {
+public class TextFieldDef extends TextBaseFieldDef implements PrefixQueryable {
+  protected PrefixFieldDef prefixFieldDef;
+  private Map<String, IndexableFieldDef<?>> childFieldsWithPrefix;
+
   public TextFieldDef(
       String name, Field requestField, FieldDefCreator.FieldDefCreatorContext context) {
     super(name, requestField, context);
+    if (requestField.hasIndexPrefixes()) {
+      if (!requestField.getSearch()) {
+        throw new IllegalArgumentException(
+            "Cannot set index_prefixes on unindexed field [" + name + "]");
+      }
+      this.prefixFieldDef = new PrefixFieldDef(getName(), requestField, context);
```

Comment:
Instead of passing `requestField`, it might be better to use it to build a simplified `Field` with just the properties you want.